### PR TITLE
Add config template and update samples

### PR DESCRIPTION
**Pull request by stonepreston**
This PR adds config templates following the specification in the [docs](https://cluster-api.sigs.k8s.io/developer/providers/bootstrap.html#bootstraptemplate-resources).

It also updates the samples to accurately reflect the defined types

### CODE REVIEW

File: api/v1beta1/charmedk8sconfigtemplate_types.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type CharmedK8sConfigConfigTemplateResource struct {
```
Comment (Cynerva) [inline api/v1beta1/charmedk8sconfigtemplate_types.go]: I think an extra `Config` got in the name here, was that intentional?
```suggestion
type CharmedK8sConfigTemplateResource struct {
```

File: api/v1beta1/charmedk8sconfigtemplate_types.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	Spec CharmedK8sConfigSpec `json:"spec"`
+}
+
+// CharmedK8sConfigTemplateSpec defines the desired state of CharmedK8sConfigTemplate
+type CharmedK8sConfigTemplateSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Template CharmedK8sConfigConfigTemplateResource `json:"template"`
```
Comment (Cynerva) [inline api/v1beta1/charmedk8sconfigtemplate_types.go]:
```suggestion
	Template CharmedK8sConfigTemplateResource `json:"template"`
```

**Review by Cynerva — CHANGES_REQUESTED** [review]
Looks good, I just have one nit

File: api/v1beta1/charmedk8sconfigtemplate_types.go (hunk 1 above)
Comment (stonepreston) [reply api/v1beta1/charmedk8sconfigtemplate_types.go]: whoops! no not intentional, thanks for that

**Review by Cynerva — APPROVED** [review]
:+1:

**Pull request merged**